tiledb/cloud/dag/dag.py

```diff
@@ -153,57 +158,68 @@ def exec(self, namespace: Optional[str] = None):
         # Override namespace if passed
         if namespace is not None and not self.local_mode:
             self.kwargs["namespace"] = namespace
-        self.status = st.Status.RUNNING
-        self.future = self.dag.executor.submit(self._do_exec)
-        assert self.future  # Needed to silence mypy.
-        self.future.add_done_callback(self._handle_completed_future)
+        if not self.status == st.Status.CANCELLED:
+            self.status = st.Status.RUNNING
+            ft = self.dag.executor.submit(self._do_exec)
+        else:
+            ft = futures.Future()
+            ft.cancel()
+        self._future = ft
+        ft.add_done_callback(self._handle_completed_future)
+
+    def _do_exec(self) -> results.Result[_T]:
+        # We have to make a shallow copy of kwargs here.
+        # Since we modify the kwargs dictionary here before passing it
+        # to the wrapped function, we need to ensure that we don't mutate
+        # our stored kwargs.
+        raw_kwargs = dict(self.kwargs)
```

Comment:
Based on the comment above, don't we need a deep copy here?

Reply:
In this case, no; our concerns are only about the arguments we add to the dictionary (`_download_results` and `stored_param_uuids`). I updated the comment to be a bit more precise here.

Reply:
Got it, ok, thanks.